### PR TITLE
doc: Bluetooth software maturity level 

### PR DESCRIPTION
**Pull request by yuxCai**
doc: update software maturity level for Bluetooth
Update software maturity level for Bluetooth.
Ref: https://github.com/nrfconnect/sdk-nrfxlib/blob/main/softdevice_controller/README.rst

Signed-off-by: Yuxuan Cai <yuxuan.cai@nordicsemi.no>

### CODE REVIEW

File: doc/nrf/releases_and_maturity/software_maturity.rst [hunk 1]
```diff
@@ -467,58 +467,352 @@ The following table indicates the software maturity levels of the support for ea
...
         - Experimental
         - Supported
+        - --
+        - --
+        - --
+        - --
+      * - **Advertiser/Peripheral**
+        - Supported
+        - Supported
+        - --
```
Comment (Tschet1) [inline doc/nrf/releases_and_maturity/software_maturity.rst:487]: 52820 should support the scanner/central role

File: doc/nrf/releases_and_maturity/software_maturity.rst [hunk 2]
```diff
@@ -467,58 +467,352 @@ The following table indicates the software maturity levels of the support for ea
...
+        - Supported
+        - Supported
+        - Experimental
+        - Supported
+        - --
+        - --
+        - --
+        - --
+      * - **Advertising Extensions**
+        - --
```
Comment (Tschet1) [inline doc/nrf/releases_and_maturity/software_maturity.rst:527]: all 52 devices should support advertising extensions

**Review by divipillai — APPROVED** [review]

File: doc/nrf/releases_and_maturity/software_maturity.rst [hunk 3]
```diff
@@ -467,58 +467,309 @@ The following table indicates the software maturity levels of the support for ea
...
+        - --
+        - --
+        - --
+
+| [1]: Concurrent central, observer, peripheral, and broadcaster roles with up to 20 concurrent connections along with one Observer and one Broadcaster (subject to RAM availability)
+| [2]: Do not support encrypting and decrypting the Isochronous Channels packets
+| [3]: Only AoA transmitter is supported
+
+The following table indicates the software maturity levels of the support for each proprietary Bluetooth feature:
```
Comment (yuxCai) [inline doc/nrf/releases_and_maturity/software_maturity.rst:685]: I will keep a separate table for the proprietary features before we know how to separate/group them in the same table

File: doc/nrf/releases_and_maturity/software_maturity.rst [hunk 4]
```diff
@@ -467,58 +467,309 @@ The following table indicates the software maturity levels of the support for ea
...
+        - Supported
         - Supported
         - Supported
         - Experimental
         - Supported
+        - --
+        - --
+        - --
+        - --
+      * - **Coded PHY (Long Range)**
```
Comment (Tschet1) [inline doc/nrf/releases_and_maturity/software_maturity.rst:484]: for the 52 series, the following devices support Coded:
- 52811
- 52820
- 52833
- 52840

File: doc/nrf/releases_and_maturity/software_maturity.rst [hunk 5]
```diff
@@ -467,58 +467,309 @@ The following table indicates the software maturity levels of the support for ea
...
+        - Supported
+        - Supported\ :sup:`2`
+        - Supported
         - Experimental
+        - Supported
+        - --
+        - --
+        - --
+        - --
+      * - **Direction Finding**\ :sup:`3`
```
Comment (Tschet1) [inline doc/nrf/releases_and_maturity/software_maturity.rst:582]: did you double-check that direction finding is not supported on 54?

Comment (Tschet1) [inline doc/nrf/releases_and_maturity/software_maturity.rst:582]: for the 52 series, the following devices support DFE:
- 52811
- 52820
- 52833

File: doc/nrf/releases_and_maturity/software_maturity.rst [hunk 6]
```diff
@@ -467,58 +467,309 @@ The following table indicates the software maturity levels of the support for ea
...
+        - --
+        - --
+        - --
+        - --
+
+  | [1]: Concurrent central, observer, peripheral, and broadcaster roles with up to 20 concurrent connections along with one Observer and one Broadcaster (subject to RAM availability)
+  | [2]: Do not support encrypting and decrypting the Isochronous Channels packets
+  | [3]: Only AoA transmitter is supported
+
+The following table indicates the software maturity levels of the support for each proprietary Bluetooth feature:
```
Comment (Tschet1) [inline doc/nrf/releases_and_maturity/software_maturity.rst:685]: For the proprietary features, should we link them here to some description?
I think this is a good starting point, but we could probably mention additional proprietary features in a second step.

**Review by Tschet1 — CHANGES_REQUESTED** [review]
The table needs to be updated.

File: doc/nrf/releases_and_maturity/software_maturity.rst (hunk 6 above)
Comment (Tschet1) [reply doc/nrf/releases_and_maturity/software_maturity.rst:685]: not sure. But we can do that later, I think.

**Review by Tschet1 — APPROVED** [review]

**Pull request merged**